launcher/src/backend/Monitoring.js

```diff
@@ -2517,6 +2517,131 @@ rm -rf diskoutput
     return serviceInfos;
   }
 
+  async getValidatorStats(validatorPublicKey) {
+    const validators_arr = [validatorPublicKey];
+    const verbose = true;
+    const proposer = false;
+
+    const beaconStatus = await this.getBeaconStatus();
+    const beaconAPIPort = beaconStatus.data[0].beacon.destinationPort;
+
+    const baseURL = `http://localhost:${beaconAPIPort}`;
+
+    const beaconAPICmdGenesisTime = `curl -s -X GET '${baseURL}/eth/v1/beacon/genesis' -H 'accept: application/json'`;
+    const genesisResShell = await this.nodeConnection.sshService.exec(beaconAPICmdGenesisTime);
+
+    let output = {};
+
+    const { genesis_time } = JSON.parse(genesisResShell.stdout).data;
+    const current_time = Math.floor(Date.now() / 1000);
+    const slot_time = 12;
+    const slot_timeout = slot_time - ((current_time - genesis_time) % slot_time);
+    const current_slot = Math.floor((current_time - genesis_time) / slot_time);
+    const current_epoch = Math.floor(current_slot / 32);
+
+    output = { currentEpoch: current_epoch };
+
```

Comment:
the last argument below ({"Content-Type": "application/json"}) is not needed since this is the default anyway